crates/solver/src/settlement_post_processing/optimize_unwrapping.rs

```diff
@@ -0,0 +1,201 @@
+use crate::settlement::Settlement;
+use anyhow::Result;
+use contracts::WETH9;
+use primitive_types::U256;
+
+pub async fn optimize_unwrapping<V, VFut, B, BFut>(
+    settlement: &mut Settlement,
```

Comment:
Since we have to clone the settlement anyway maybe it would be nicer to take and return it by value. That way the caller can't ignore the result so easily and it is clearer under what conditions the settlement changes. With the current signature it might not be clear whether the settlement will be modified if the function returns Err.

Reply:
I used an out parameter to leave the original `Settlement` untouched while also reporting an error. With `-> Result<Settlement>` we would loose the original settlement in case of an error and with `-> Settlement` we would not be able to report any errors at all.
Because there is currently no good way to handle an error anyway, I could simply drop the error handling and always return a settlement.

Reply:
We would have to clone it to not lose it, yes (or return it in the Error). I do think this way is fine too so no need to change it. Maybe add a comment saying that settlement is not modified if the Result is Err.

Reply:
I decided to drop the error propagation after all.